src/cljs/main/broadfcui/page/workspace/data/import_data.cljs

```diff
@@ -69,17 +69,19 @@
    :do-upload
    (fn [{:keys [props state]}]
      (swap! state assoc :loading? true)
-     (endpoints/call-ajax-orch
-      {:endpoint ((if (= "data" (:import-type props))
-                    endpoints/import-entities
-                    endpoints/import-attributes)
-                  (:workspace-id props))
-       :raw-data (utils/generate-form-data {(if (= "data" (:import-type props)) :entities :attributes) (:file @state)})
-       :encType "multipart/form-data"
-       :on-done (fn [{:keys [success? get-parsed-response]}]
-                  (swap! state dissoc :loading? :file :file-contents)
-                  (if success?
-                    (do
-                      (swap! state assoc :upload-result {:success? true})
-                      ((:on-data-imported props)))
-                    (swap! state assoc :upload-result {:success? false :error (get-parsed-response false)})))}))})
+     (if-let [on-upload (:on-upload props)]
+       (on-upload (select-keys @state [:file-contents]))
+       (endpoints/call-ajax-orch
+          {:endpoint ((if (= "data" (:import-type props))
+                        endpoints/import-entities
+                        endpoints/import-attributes)
+                      (:workspace-id props))
+           :raw-data (utils/generate-form-data {(if (= "data" (:import-type props)) :entities :attributes) (:file @state)})
+           :encType "multipart/form-data"
+           :on-done (fn [{:keys [success? get-parsed-response]}]
+                      (swap! state dissoc :loading? :file :file-contents)
+                      (if success?
+                        (do
+                          (swap! state assoc :upload-result {:success? true})
+                          ((:on-data-imported props)))
+                        (swap! state assoc :upload-result {:success? false :error (get-parsed-response false)})))})))})
```

Comment:
A couple of things in here can be moved into a `let` - 

`(= "data" (:import-type props)`, `:file-contents` and `:file` from state, and `:workspace-id` and `:on-data-imported` from props

I think the let could be inside the `if-let`, so that way we don't set those unless the `if-let` proves true

Reply:
@asingh7115 Updated. Please re-review carefully... these improvements were in code that was related but not in the code path I was modifying. It should be a straightforward refactor but we need to be extra careful that I didn't mess anything up. Thanks!